### PR TITLE
test(core-lib): add u256 regression vectors

### PR DESCRIPTION
**Pull request by apetro2**
## Describe your changes                                                                                                                                                                        
                                                                                                                                                                                                  
  Follow-up to #2891.
                                                                                                                                                                                                  
  This adds LE-oriented regression vectors for the current `miden::core::math::u256` API in `core-lib` tests. It introduces a small Rust reference model backed by `{ lo: u128, hi: u128 }` and uses it to verify the existing MASM procedures against fixed boundary cases and a pseudorandom pair.                                                                                            

  The goal is to make the current behavior explicit in tests before #2641 lands. This keeps the change tightly scoped to test coverage while giving future `u256` endian/layout refactors a concrete regression guard.
                                                                                                                                                                                                  
  Validation:                                                                                                                                                                                     
  - `cargo test -p miden-core-lib u256_mod -- --nocapture`                                                                                                                                        
  - `cargo clippy -p miden-core-lib --tests -- -W clippy::all`

  ## Checklist before requesting a review                                                                                                                                                         
  - [x] Repo forked and branch created from `next` according to naming convention.
  - [x] Commit messages and codestyle follow [conventions](../CONTRIBUTING.md).                                                                                                                   
  - [x] Commits are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
  - [x] Relevant issues are linked in the PR description.                                                                                                                                         
  - [x] Tests added for new functionality                                                                                                                                                         
  - [ ] Documentation/comments updated according to changes.                                                                                                                                      
  - [ ] Updated `CHANGELOG.md'

### CODE REVIEW

File: crates/lib/core/tests/math/u256_mod.rs [hunk 1]
```diff
@@ -1,27 +1,15 @@
-use miden_utils_testing::{proptest::prelude::*, rand::rand_vector};
-use num_bigint::BigUint;
+use core::ops::{BitAnd, BitOr, BitXor};
```
Comment (huitseeker) [inline crates/lib/core/tests/math/u256_mod.rs:1]: Since this drops the last `BigUint` / `num_bigint` use I can find under `crates/lib/core`, should `num-bigint` also come out of `crates/lib/core/Cargo.toml`?

File: crates/lib/core/tests/math/u256_mod.rs [hunk 2]
```diff
@@ -121,40 +147,219 @@ proptest! {
...
+        U256::from_bit(191),
+        U256::from_bit(255),
+        pseudo_random_pair().0,
+        pseudo_random_pair().1,
+    ]
+}
+
+fn regression_pairs() -> Vec<(U256, U256)> {
+    let (rand_a, rand_b) = pseudo_random_pair();
+    vec![
```
Comment (huitseeker) [inline crates/lib/core/tests/math/u256_mod.rs:312]: I think this set still misses the add overflow path. None of these pairs seem to make `a + b >= 2^256`, so `wrapping_add`, `overflowing_add`, and `widening_add` only check the `carry = 0` case. Could we add one pair like `U256::new(u128::MAX, u128::MAX)` plus `U256::from_bit(0)`?

**Review by huitseeker — CHANGES_REQUESTED** [review]
Please add a Changelog line!

**Comment by apetro2** [comment]
@huitseeker Done, appreciate your review!

**Review by huitseeker — APPROVED** [review]
LGTM!

**Comment by huitseeker** [comment]
Closing in favor of #3026
